### DISCUSSION TITLE
STORAGE OF OBSJECT IN MULTIFAB

### DISCUSSION THREAD

**Discussion by FernandoIGuerra** [AMReX-Codes/amrex#3999] [Q&A]
Hi Everyone,
is it possible to allocate an object (class) at each component of the MultiFab. So far, I implemented the code shown below. However, the compiler indicates that there is an conversion error from Array4 to non-scalar Array4<LEMElement> The intuitive idea behind this is ti create LEM elements inside each LES cell. Also, I was wondering if the allocation of objects would cause any problem in ParallelFor. Thanks in advance.
```c++
template<typename T>
class LEMElement {
private:
    T* primitive_variables;
    T* position;
    size_t primitive_variables_size;
    size_t position_size;

amrex::MultiFab lem_mf(ba, dm, num_lem_elements, 0);

        for(amrex::MFIter mfi(lem_mf); mfi.isValid(); ++mfi){
                	const amrex::Box& bx = mfi.validbox();
                	const amrex::Array4<LEMElement<double>>mf_array = lem_mf.array(mfi);
                	amrex::ParallelFor(bx, [=] AMREX_GPU_DEVICE(int i, int j, int k){
                		mf_array(i,j,k, 0) = LEMElement<double>(primitive_size, position_size);
                	});
                }
```

**Comment by smahesh14** [comment]
MultiFab is 'FabArray<BaseFab<Real>>'; for storing objects, you can use 'FabArray<BaseFab<T>>' where T can be a class.
